ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -487,28 +488,31 @@ boolean isDataNodeInFullAutoMode(DataNodeId dataNodeId, boolean shouldCheckPrope
       logger.trace("DataNode {} doesn't have any tags", instanceName);
       return false;
     }
-    // Before turn on FULL_AUTO, we have to make sure each host only stores partitions from one resource.
-    // Meaning there should be only one tag in the list
-    if (tags.size() != 1) {
-      logger.trace("DataNode {} has multiple tags {}", instanceName, tags);
-      return false;
-    }
-    // Make sure all the resources turned on FULL_AUTO mode
-    String tag = tags.iterator().next();
-    ResourceProperty property = dcToTagToResourceProperty.get(dcName).get(tag);
-    if (property == null) {
-      return false;
-    }
-    // Return true if either the resource is in Full-auto or we are rolling back from Full-auto to Semi-auto
-    if (property.rebalanceMode.equals(IdealState.RebalanceMode.FULL_AUTO)) {
-      return true;
-    }
-    // If we are here, then ResourceProperty shows RebalanceMode as NOT FULL_AUTO.
-    if (!shouldCheckPropertyStore) {
-      return false;
+    logger.trace("Instance {} has tags {}", instanceName, tags);
+    // Check if all the resources that this host belongs to are in FULL_AUTO mode
+    for (String tag : tags) {
+      ResourceProperty property = dcToTagToResourceProperty.get(dcName).get(tag);
+      if (property == null) {
+        logger.trace("No Resource property found for tag {}", tag);
+        return false;
+      }
+      // Return false if any of the resource is in semi-auto
+      if (!property.rebalanceMode.equals(IdealState.RebalanceMode.FULL_AUTO)) {
+        logger.trace("Resource with tag {} is in semi-auto mode", tag);
+        return false;
+      }
+
+      if (shouldCheckPropertyStore) {
+        // Return false if any of the resource is not present in Full-Auto migration list
+        ZNRecord zNRecord =
+            helixPropertyStoreInLocalDc.get(FULL_AUTO_MIGRATION_ZNODE_PATH, null, AccessOption.PERSISTENT);
+        if (zNRecord == null || !zNRecord.getListField(RESOURCES_STR).contains(property.name)) {
+          logger.trace("Resource with tag {} is in not present in /AdminConfigs/FullAutoMigration list", tag);
+          return false;
+        }
+      }
```

Comment:
the for loop isn't quite right, we shouldn't have to check property store multiple times. The property store checking should happen after for loop when there is at least one tag/resource that is not on FULL AUTO.

Reply:
Sorry my bad. Thank you for catching it. Updated the condition.